src/Stage/Parse/Parser.elm

```diff
@@ -1450,6 +1451,20 @@ case_ config =
         |> located
 
 
+shader : ExprConfig -> Parser_ LocatedExpr
+shader config =
+    let
+        startToken =
+            "[glsl|"
+    in
+    P.succeed ()
+        |. P.symbol (P.Token startToken ShaderProblem)
+        |. P.chompUntil (P.Token "|]" ShaderProblem)
+        |> P.getChompedString
+        |> P.map (String.dropLeft (String.length startToken) >> Frontend.Shader)
```

Comment:
I believe we could get rid of the `dropLeft` postprocessing by instead chomping only after that prefix.

I'm skipping boilerplate because I'm writing this on my phone, but the idea would be:

```elm
P.succeed Frontend.Shader
|. P.symbol "[glsl|"
|= P.getChompedString (P.chompUntil "|]")
|. P.symbol "|]"
|> located
```